### ISSUE TITLE
pip not works after installing angr

### ISSUE THREAD

**Issue by agfn** [angr/angr#2378]
---

**Describe the bug.**

After installing, pip not works.

```shell
# installing process
mkvirtualenv -p $(which python3)  angr
python -m pip install angr
```

All works as normal, but pip.
```
(angr) ➜  Downloads python -m pip list
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pip/__main__.py", line 19, in <module>
    sys.exit(_main())
  File "/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pip/_internal/cli/main.py", line 73, in main
    command = create_command(cmd_name, isolated=("--isolated" in cmd_args))
  File "/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pip/_internal/commands/__init__.py", line 96, in create_command
    module = importlib.import_module(module_path)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pip/_internal/commands/list.py", line 13, in <module>
    from pip._internal.cli.req_command import IndexGroupCommand
  File "/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pip/_internal/cli/req_command.py", line 20, in <module>
    from pip._internal.operations.prepare import RequirementPreparer
  File "/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pip/_internal/operations/prepare.py", line 16, in <module>
    from pip._internal.distributions import (
  File "/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pip/_internal/distributions/__init__.py", line 2, in <module>
    from pip._internal.distributions.wheel import WheelDistribution
  File "/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pip/_internal/distributions/wheel.py", line 5, in <module>
    from pip._internal.utils.wheel import pkg_resources_distribution_for_wheel
  File "/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pip/_internal/utils/wheel.py", line 12, in <module>
    from pip._vendor.six import PY2, ensure_str
ImportError: cannot import name 'ensure_str' from 'six' (/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/six.py)
```

**Environment Information.**

```
angr environment report
=============================
Date: 2020-11-06 20:53:33.933289
Running in virtual environment at /home/lt/.virtualenvs/angr
Platform: linux-x86_64
Python version: 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0]
######## angr #########
Python found it in /home/lt/.virtualenvs/angr/lib/python3.8/site-packages/angr
Pip version angr 9.0.4663
Couldn't find git info
######## ailment #########
Python found it in /home/lt/.virtualenvs/angr/lib/python3.8/site-packages/ailment
Pip version ailment 9.0.4663
Couldn't find git info
######## cle #########
Python found it in /home/lt/.virtualenvs/angr/lib/python3.8/site-packages/cle
Pip version cle 9.0.4663
Couldn't find git info
######## pyvex #########
Python found it in /home/lt/.virtualenvs/angr/lib/python3.8/site-packages/pyvex
Pip version pyvex 9.0.4663
Couldn't find git info
######## claripy #########
Python found it in /home/lt/.virtualenvs/angr/lib/python3.8/site-packages/claripy
Pip version claripy 9.0.4663
Couldn't find git info
######## archinfo #########
Python found it in /home/lt/.virtualenvs/angr/lib/python3.8/site-packages/archinfo
Pip version archinfo 9.0.4663
Couldn't find git info
######## z3 #########
Python found it in /home/lt/.virtualenvs/angr/lib/python3.8/site-packages/z3
Pip version z3-solver 4.8.9.0
Couldn't find git info
######## unicorn #########
Python found it in /home/lt/.virtualenvs/angr/lib/python3.8/site-packages/unicorn
Pip version unicorn 1.0.2rc4
Couldn't find git info
######### Native Module Info ##########
angr: <CDLL '/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/angr/lib/angr_native.so', handle 16f82d0 at 0x7f54fe834e80>
unicorn: <CDLL '/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/unicorn/lib/libunicorn.so', handle e5f560 at 0x7f54fda2fbe0>
pyvex: <cffi.api._make_ffi_library.<locals>.FFILibrary object at 0x7f54fe77d520>
z3: <CDLL '/home/lt/.virtualenvs/angr/lib/python3.8/site-packages/z3/lib/libz3.so', handle 10b7ff0 at 0x7f54fbb2a8e0>
```

**To Reproduce.**

**Additional context.**

**Comment by agfn** [comment]
Seems that six version is not right. Solved by installing six=1.15.0.

**Comment by rhelmot** [comment]
probably caused by https://github.com/pysmt/pysmt/issues/652
